apps/meteor/tests/unit/server/publications/messages.spec.ts

```diff
@@ -0,0 +1,282 @@
+import type { IMessage } from '@rocket.chat/core-typings';
+import { expect } from 'chai';
+import { describe, it } from 'mocha';
+import proxyquire from 'proxyquire';
+import sinon from 'sinon';
+
+const mockMeteorError = class extends Error {
+	constructor(public error: string, public reason: string, public details: any) {
+		super(reason);
+		this.name = 'Meteor.Error';
+	}
+};
+
+const messagesMock = {
+	findForUpdates: sinon.stub(),
+	trashFindDeletedAfter: sinon.stub(),
+	trashFind: sinon.stub(),
+};
+
+const {
+	extractTimestampFromCursor,
+	mountCursorQuery,
+	mountCursorFromMessage,
+	mountNextCursor,
+	mountPreviousCursor,
+	handleWithoutPagination,
+	handleCursorPagination,
+} = proxyquire.noCallThru().load('../../../../server/publications/messages', {
+	'meteor/check': {
+		check: sinon.stub(),
+	},
+	'meteor/meteor': {
+		'Meteor': {
+			methods: sinon.stub(),
+			Error: mockMeteorError,
+		},
+		'@global': true,
+	},
+	'@rocket.chat/models': {
+		Messages: messagesMock,
+	},
+});
+
+describe('extractTimestampFromCursor', () => {
+	it('should return the correct timestamp', () => {
+		const cursor = new Date('2024-01-01T00:00:00.000Z').getTime().toString();
+		const timestamp = extractTimestampFromCursor(cursor);
+		expect(timestamp).to.be.an.instanceOf(Date);
+		expect(timestamp.getTime()).to.equal(parseInt(cursor, 10));
+	});
+});
+
+describe('mountCursorQuery', () => {
+	const mockDate = new Date('2024-01-01T00:00:00.000Z').getTime();
+
+	it('should return a query with $gt when next is provided', () => {
+		const result = mountCursorQuery({ next: mockDate.toString() });
+		expect(result.query).to.deep.equal({ $gt: new Date(mockDate) });
+	});
+
+	it('should return a query with $lt when previous is provided', () => {
+		const result = mountCursorQuery({ previous: mockDate.toString() });
+		expect(result.query).to.deep.equal({ $lt: new Date(mockDate) });
+	});
+
+	it('should return a query with $gt and sort when neither next nor previous is provided', () => {
```

Comment:
I think this is not possible due to the higher layers validation, but what if both are provided? Maybe put a validation here as well if we want to use the same approach in other places.